src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -264,8 +266,12 @@ float64_t CCrossValidation::evaluate_one_run()
 		m_machine->set_store_model_features(true);
 
 		/* do actual cross-validation */
+#pragma omp parallel for
 		for (index_t i=0; i <num_subsets; ++i)
 		{
+			CMachine* machine=(CMachine*)m_machine->clone();
+			SG_REF(machine);
+			machine->set_store_model_features(true);
 			/* evtl. update xvalidation output class */
 			CCrossValidationOutput* current=(CCrossValidationOutput*)
```

Comment:
keep in mind that your x-validation output currently is shared across the threads.
That is fine actually (updating it is fast), but you might want to do the update of it atomic so that the different threads dont conflict here